lib/ansible/plugins/action/docker_service.py

```diff
@@ -0,0 +1,182 @@
+from ansible.utils.hashing import checksum
+from ansible.utils.unicode import to_bytes, to_str
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+        ''' handler for file transfer operations '''
+        if task_vars is None:
+            task_vars = dict()
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+
+        source  = self._task.args.get('project_src', None)
+        dest    = self._task.args.get('remote_src', None)
+        raw     = False
+        force   = True
+        follow  = True
+
+        if source and dest:
+
+            # Check if the source ends with a "/"
+            source_trailing_slash = False
+            if source:
+                source_trailing_slash = self._connection._shell.path_has_trailing_slash(source)
+
+            try:
+                source = self._find_needle('files', source)
+            except AnsibleError as e:
+                result['failed'] = True
+                result['msg'] = to_str(e)
+                return result
+
+            # A list of source file tuples (full_path, relative_path) which will try to copy to the destination
+            source_files = []
+
+            # If source is a directory populate our list else source is a file and translate it to a tuple.
+            if os.path.isdir(to_bytes(source, errors='strict')):
+                # Get the amount of spaces to remove to get the relative path.
+                if source_trailing_slash:
+                    sz = len(source)
+                else:
+                    sz = len(source.rsplit('/', 1)[0]) + 1
+
+                # Walk the directory and append the file tuples to source_files.
+                for base_path, sub_folders, files in os.walk(source):
+                    for file in files:
+                        full_path = os.path.join(base_path, file)
+                        rel_path = full_path[sz:]
+                        if rel_path.startswith('/'):
+                            rel_path = rel_path[1:]
+                        if rel_path.startswith(os.path.basename(source) + '/'):
+                            rel_path = rel_path.replace(os.path.basename(source) + '/', '', 1)
+                        source_files.append((full_path, rel_path))
+
+                # If it's recursive copy, destination is always a dir,
+                # explicitly mark it so (note - copy module relies on this).
+                if not self._connection._shell.path_has_trailing_slash(dest):
+                    dest = self._connection._shell.join_path(dest, '')
+            else:
+                source_files.append((source, os.path.basename(source)))
+
+            changed = False
+            module_return = dict(changed=False)
+
+            # If this is a recursive action create a tmp path that we can share as the _exec_module create is too late.
+            remote_user = task_vars.get('ansible_ssh_user') or self._play_context.remote_user
```

Comment:
you should NOT use ansible_ssh_user directly, it might be the wrong var, play_context.remote_user should always be the correct one.